sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -712,15 +741,14 @@ void cleanup(PipelineOptions options) throws Exception {
       @Override
       public void populateDisplayData(DisplayData.Builder builder) {
         super.populateDisplayData(builder);
-        TableReference table = getTable();
+        ValueProvider<TableReference> table = getTableProvider();
 
         if (table != null) {
-          builder.add(DisplayData.item("table", toTableSpec(table))
+          builder.add(DisplayData.item("table", table.isAccessible()
+                  ? toTableSpec(table.get()) : table.toString())
             .withLabel("Table"));
         }
-        String queryString = query == null
-            ? null : query.isAccessible()
-            ? query.get() : query.toString();
+        String queryString = query == null ? null : tableProvider(query);
```

Comment:
I think we could add this to the DisplayDataBuilder. Basically, if it receives a ValueProvider it can do:

1. If it is not accessible, include the ValueProvider#toString()
2. If it is accessible, call `ValueProvider#get`, and then add that (using all of the normal logic already there).